releaselock.php

```diff
@@ -0,0 +1,40 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Release staled lock.
+ *
+ * @package    tool_lockstats
+ * @author     Ilya Tregubov <ilyatregubov@catalyst-au.net>
+ * @copyright  2019 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+
+require_once(dirname(__FILE__) . '/../../../config.php');
+require_once($CFG->libdir . '/adminlib.php');
+
+require_login(null, false);
+require_capability('moodle/site:config', context_system::instance());
+
+if (isset($_GET['resourcekey'])) {
+    $resourcekey = $_GET['resourcekey'];
+    $cronlockfactory = \core\lock\lock_config::get_lock_factory('cron');
+    $lock = $cronlockfactory->get_lock($resourcekey, 0);
```

Comment:
If you gain a lock then you must release it, so check that it exists.

IF you gain the lock, then redirect to the index page.

IF NOT then the lock is still valid, so show a message like 'Lock still in use by another process '. 

This second message could just be shown at the top of the locks_detail.php in which case I would merge releaselock.php into locks_detail.php